apps/server/src/shared/repo/base.do.repo.ts

```diff
@@ -79,7 +79,7 @@ export abstract class BaseDORepo<DO extends BaseDO, E extends BaseEntity, P> {
 		});
 	}
 
-	async deleteById(id: string | string[]): Promise<number> {
+	async deleteById(id: EntityId | EntityId[]): Promise<number> {
```

Comment:
We should only allow deleting domain objects by DO reference and not by id. This general method would then become more like `delete(do: DO | DO[])`. In our services and use cases we should deal with real domain objects in order to ensure the business requirements are met. If we want to delete one ore more objects we should pass them as objects to the repository like we are doing it e.g. in the `save` method(s).

Also we should not use `em.nativeDelete` in our repositories if possible because that doesn't use the Unit of Work, cascading etc. It's better to use `em.remove()` instead.

However, there a situations where we have to process lots of database entries at once (bulk operations) and we need something like `em.nativeDelete()`. But these operations should not be implemented in our repositories but in other places (like scripts).

Reply:
Yes. It has been discussed and therefore should be changed. https://docs.dbildungscloud.de/display/DBH/ARC+Meeting+2022-12-01

Reply:
The appointment was yesterday.. We didn't plan the effort at the start of the sprint. The decision would entail even more changes. But thanks you for sharing the decision in this point :)

Reply:
Alright. We can move that to the next sprint.